lib/ruby_lsp/global_state.rb

```diff
@@ -166,8 +168,24 @@ def detect_typechecker(dependencies)
     end
 
     sig { returns(T::Boolean) }
-    def bin_rails_present
-      File.exist?(File.join(workspace_path, "bin/rails"))
+    def rails_app?
+      config = rails_app_config
+      return false unless config
+
+      result = Prism.parse(config)
+      return false unless result
+
+      dispatcher = Prism::Dispatcher.new
+      listener = Listeners::RailsApp.new(dispatcher)
+      dispatcher.dispatch(result.value)
+
+      listener.rails_app
```

Comment:
I would probably have done it with a simple regex, but your solution looks much cleaner 👌🏼